diagnostics/stratonovich_additive.py

```diff
@@ -0,0 +1,166 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import tqdm
+from scipy import stats
+
+from tests.basic_sde import AdditiveSDE
+from torchsde import sdeint, BrownianInterval
+from torchsde.settings import LEVY_AREA_APPROXIMATIONS
+from .utils import to_numpy, makedirs_if_not_found, compute_mse
+
+from tests.problems import Ex3Additive
+
+def inspect_sample():
+    batch_size, d, m = 32, 1, 5
+    steps = 10
+
+    ts = torch.linspace(0., 5., steps=steps, device=device)
+    dt = 3e-1
+    y0 = torch.ones(batch_size, d, device=device)
+    sde = AdditiveSDE(d=d, m=m, sde_type='stratonovich').to(device)
+    sde_ito = AdditiveSDE(d=d, m=m, sde_type='ito').to(device)
+    sde_ito.f_param = sde.f_param
+    sde_ito.g_param = sde.g_param
+
+    with torch.no_grad():
+        bm = BrownianInterval(t0=ts[0], t1=ts[-1], shape=(batch_size, m), dtype=y0.dtype, device=device,
+                              levy_area_approximation=LEVY_AREA_APPROXIMATIONS.space_time)
+
+        ys_heun = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='heun')
+        ys_euler_heun = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='euler_heun')
+        ys_midpoint = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='midpoint')
+        ys_milstein_strat = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='milstein')
+        ys_mil_strat_grad_free = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='milstein',
+                                        options={'grad_free': True})
+        ys_true = sdeint(sde_ito, y0=y0, ts=ts, dt=1e-3, bm=bm, method='euler')
+
+        print(ys_heun.shape, ys_heun.squeeze().shape)
+
+        ys_heun = ys_heun.squeeze().t()
+        ys_euler_heun = ys_euler_heun.squeeze().t()
+        ys_midpoint = ys_midpoint.squeeze().t()
+        ys_milstein_strat = ys_milstein_strat.squeeze().t()
+        ys_mil_strat_grad_free = ys_mil_strat_grad_free.squeeze().t()
+        ys_true = ys_true.squeeze().t()
+
+        ts_, ys_heun_, ys_euler_heun_, ys_midpoint_, ys_milstein_strat_, ys_mil_strat_grad_free_, ys_true_ = to_numpy(
+            ts, ys_heun, ys_euler_heun, ys_midpoint, ys_milstein_strat, ys_mil_strat_grad_free, ys_true)
+
+    # Visualize sample path.
+    img_dir = os.path.join('.', 'diagnostics', 'plots', 'stratonovich_additive')
+    makedirs_if_not_found(img_dir)
+
+    for i, (ys_heun_i, ys_euler_heun_i, ys_midpoint_i, ys_milstein_strat_i, ys_mil_strat_grad_free_i, ys_true_i) in enumerate(
```

Comment:
Line too long.

Reply:
This line is 120 - should I wrap it anyway?